### PR TITLE
Implemented user internal interests store and replaced user deliverable categories store

### PR DESCRIPTION
**Pull request by tommylau523**
Most of the changes are renames

### CODE REVIEW

**Comment by tommylau523** [comment]
* **#2361** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2361?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2360** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2360?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2359** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2359?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2358** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2358?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2347** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2347?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#2348](https://github.com/terraware/terraware-server/pull/2348) <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2348?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tommylau523 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/main/kotlin/com/terraformation/backend/customer/db/UserInternalInterestsStore.kt [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          .select(INTERNAL_INTEREST_ID)
+          .from(this)
+          .where(USER_ID.eq(userId))
+          .fetchSet(INTERNAL_INTEREST_ID.asNonNullable())
+    }
+  }
+
+  /**
+   * Returns a condition that can be added to a query on the users table to restrict the results to
+   * users with a particular deliverable category or with no deliverable categories at all.
```
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/customer/db/UserInternalInterestsStore.kt]: "Deliverable category" is still mentioned here.

File: src/main/resources/db/migration/0300/V303__UserInternalInterests.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
+-- Create user_internal_interests table
+CREATE TABLE accelerator.user_internal_interests (
+    user_id BIGINT NOT NULL REFERENCES users,
+    internal_interest_id INTEGER NOT NULL REFERENCES accelerator.internal_interests ON DELETE CASCADE,
+    created_by BIGINT NOT NULL REFERENCES users,
+    created_time TIMESTAMP WITH TIME ZONE NOT NULL,
+
+PRIMARY KEY (user_id, internal_interest_id)
```
Comment (sgrimm) [inline src/main/resources/db/migration/0300/V303__UserInternalInterests.sql]: Indent this for consistency.

File: src/main/resources/db/migration/0300/V303__UserInternalInterests.sql [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+);
+
+-- Move data from user_deliverable_categories  to user_internal_interests
+INSERT INTO accelerator.user_internal_interests (user_id, internal_interest_id, created_by, created_time)
+SELECT user_id, deliverable_category_id as internal_interest_id, created_by, created_time
```
Comment (sgrimm) [inline src/main/resources/db/migration/0300/V303__UserInternalInterests.sql]: You shouldn't need the `as internal_interest_id` here (the values are positional, not named).

**Review by sgrimm — APPROVED** [review]
Nice!

**Review by sgrimm — APPROVED** [review]

**Comment by tommylau523** [comment]
### Merge activity

* **Sep 3, 8:41 PM EDT**: @tommylau523 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2360).
* **Sep 3, 9:05 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2360) rebased this pull request as part of a merge.
* **Sep 3, 9:14 PM EDT**: @tommylau523 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2360).

**Pull request merged**